src/core/vector/qgsvectorlayer.cpp

```diff
@@ -5921,6 +5921,11 @@ QString QgsVectorLayer::htmlMetadata() const
 
   if ( isSpatial() )
   {
+    // geom column name
+    if ( const QgsVectorDataProvider *provider = dataProvider() )
+    {
+      myMetadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Geometry name" ) + QStringLiteral( "</td><td>" ) + provider->geometryColumnName() + QStringLiteral( "</td></tr>\n" );
```

Comment:
And could we call it "geometry column name"? I find it more explicit than "geometry name".